lib/plugins/aws/package/compile/events/sns/index.js

```diff
@@ -218,6 +222,37 @@ class AwsCompileSNSEvents {
                 },
               },
             });
+
+            if (event.sns.redrivePolicy && event.sns.redrivePolicy.deadLetterTargetArn) {
+              const queuePolicyLogicalId = this.provider.naming.getQueueLogicalId(functionName, `${topicName}DLQ`);
+              _.merge(template.Resources, {
```

Comment:
Let's rely on `Object.assign`

Reply:
ok! I used merge because it was alredy used in the previous code. Will try to update the rest asap

Reply:
I have created a unit test following the examples that I've seen on the same folder, but I am not sure about how to mock the "failure" of messages sent from sns to lambda. 
For my personal testing, what I did was to point the SNS subscription to an inexisting lambda as suggested here https://vgaltes.com/post/serverless-patterns-implemented-sls-dlq/ , so I  could verify that every time a message was published, a message would end in the DLQ.

Any ideas on how to do that  following the integration test example?